diag_manager/fms_diag_object.F90

```diff
@@ -552,6 +556,8 @@ logical function fms_diag_accept_data (this, diag_field_id, field_data, mask, rm
   if ((present(is_in) .and. present(ie_in)) .or. (present(js_in) .and. present(je_in))) &
     has_halos = .true.
 
+  if(has_halos) call this%FMS_diag_fields(diag_field_id)%set_halo_present()
+
```

Comment:
This is already done inside the OMP critical section:
https://github.com/NOAA-GFDL/FMS/blob/0c2f6802c531ae3c4c48c249c96d4b803a61fbdd/diag_manager/fms_diag_object.F90#L604-L656

Reply:
removed with da79b88